src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_NUMBER_OF_BUCKETS = 16;
+    static final float DEFAULT_LOADER_FACTOR = 0.75f;
+    private int numberOfBuckets = DEFAULT_NUMBER_OF_BUCKETS;
+    private int size;
+    private int index;
+
+    private Node[] buckets = new Node[DEFAULT_NUMBER_OF_BUCKETS];
+
+    private class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size + 1 > buckets.length * DEFAULT_LOADER_FACTOR) {
+            resize();
+        }
+        index = hashCode(key) % buckets.length;
```

Comment:
dont call the method hashCode() 3 times just store the value

but also this field in Node class is not used in your code - so remove it